interview-questions.md

```diff
@@ -7,60 +7,78 @@ Then, research the question to expand on your answer. Even if you feel you have
+  
+  Git: free open-source software distributed version control system, installed locally on the system, maintained by Linux Foundation, 
 
+  Github: web-based hosting service for git repositories, lives on the cloud, acquired by Microsoft in 2018 
 
 
-2. What types of evaluations will return a Boolean value?
 
-  Your answer:
+2. What types of evaluations will return a Boolean value?
 
-  Researched answer:
+  Your answer: Relational, logical, and equality operators return Boolean values.
 
+  Researched answer: 
+  We can return Booleans using Equality operators ( == === !== !=), Relational operators ( > >= < <= ), and Logical operators (and &&, or ||, and not !).
+  We can  also return a Boolean by using the method Boolean() which is used to get the Boolean value of other types of values in terms of true or false. Specifically, falsy values are considered 0, -0, null, false, NaN, undefined, or (" ") empty string and anything other than those falsy values are considered truthy.
 
 
 3. What is an index? What is the difference between index and value?
 
-  Your answer:
+  Your answer: An index is the equivalent of a location within a specified value. A value is what something has been assigned to. Indexing starts at 0 (with -1 as an index does not exist) and references the location of a value where a value could be any type of number, boolean, string, etc that is assigned or inputted.
 
-  Researched answer:
+  Researched answer: An index is a property that is read only. It contains the position of a element and can only be an integer. A value, specifically referring to data, is referring to a primitive value that the object wrapper contains. In reference to variable or property, it can either be a primitive or vaue or an object reference.
 
 
 
 4. What is a built-in method?
 
-  Your answer:
+  Your answer: A built-in method is a function that is used so often, that they've built it into the language. 
 
-  Researched answer:
+  Researched answer: A built-in method or aso referred to as a built-in function is a function that is already set up in a program or framework that has a set of statements to perform the same task when called upon each time.
 
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  Your answer: In pair programming, there is a driver and a navigator. The driver is simply the person who types out the code and the navigator is the one who vocalizes the logic and says what code the driver should type. Proper technique would be to switch every 25 minutes and to take breaks. The navigator is liable for typos.
 
-  Researched answer:
+  Researched answer: Pair programming is where two programmers work on the same code on a single workstation. One is the driver who writes the code and the other is considered as the observer or navigator who reviews each line of code as it is typed. Every few minutes or so, the programmers should switch.
```

Comment:
Good stuff, The navigator is also in charge of the general direction the driver is heading and the research involved in it.